### DISCUSSION TITLE
Reading Delf3D Flow NetCDF output into parcels

### DISCUSSION THREAD

**Discussion by VeckoTheGecko** [Parcels-code/Parcels#1204] [Q&A] [answered]
I've been working with [Delft3D Flow](https://content.oss.deltares.nl/delft3d/manuals/Delft3D-FLOW_User_Manual.pdf) simulations which have an option to export to NetCDF files. The format of the output NetCDF of `Delf3D Flow` simulations and the input of `parcels` flowfields are different.

Are there existing functions (within the codebase, or external tools) to wrangle flow data from `Delf3D Flow` simulations into the `parcels` format?

If not, I may develop these functions. Would this be a suitable fit to have somewhere in the `parcels` codebase?

**Comment by erikvansebille** [comment, marked as answer]
Thanks @VeckoTheGecko! No, we don't have out-of-the-box support for Delft3D NetCDF files yet. You could try to write a function like `FieldSet.from_nemo()`? See https://github.com/OceanParcels/parcels/blob/master/parcels/fieldset.py#L425-L499

I know that Abdel Nnafie has worked with Parcels and Delft3d; but not sure if his code is online. You could try to contact him?

**Reply by VeckoTheGecko** [reply]
Sounds good. I've just created an issue for it, happy to work on it over the next couple weeks (unless my work takes me in a different direction).

Might reach out to Abdel, but from an initial look it doesn't look too bad :).

Also, are there guidelines for contributing to `parcels`? (eg. instructions on how to install it in an [interactive way](https://stackoverflow.com/a/59667164/15545258); not sure if it would be different with `parcels` as it uses Anaconda)

**Reply by VeckoTheGecko** [reply]
All good. Found the developer install instructions in the [installation section of the website](https://oceanparcels.org/#installing).
